File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/presence/manipulative/ManipulativePresenceData.java
package aurocosh.divinefavor.common.custom_data.player.data.presence.manipulative;

import aurocosh.divinefavor.common.config.common.ConfigPresence;
import aurocosh.divinefavor.common.lib.IncrementalChanceCounter;

public class ManipulativePresenceData {
    private IncrementalChanceCounter chanceCounter;

    public ManipulativePresenceData() {
        chanceCounter = new IncrementalChanceCounter(ConfigPresence.manipulativePresence.startingChance, ConfigPresence.manipulativePresence.chanceIncrease);
    }

    public void reset() {
        chanceCounter.reset();
    }

    public boolean tryLuck() {
        return chanceCounter.tryLuck();
    }

    public float getChance() {
        return chanceCounter.getChance();
    }

    public void setChance(float chance) {
        chanceCounter.setChance(chance);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/player/data/presence/manipulative/ManipulativePresenceData.kt
package aurocosh.divinefavor.common.custom_data.player.data.presence.manipulative

import aurocosh.divinefavor.common.config.common.ConfigPresence
import aurocosh.divinefavor.common.lib.IncrementalChanceCounter

class ManipulativePresenceData {
    private val chanceCounter: IncrementalChanceCounter = IncrementalChanceCounter(ConfigPresence.manipulativePresence.startingChance, ConfigPresence.manipulativePresence.chanceIncrease)

    var chance: Float
        get() = chanceCounter.getChance()
        set(chance) = chanceCounter.setChance(chance)

    fun reset() {
        chanceCounter.reset()
    }

    fun tryLuck(): Boolean {
        return chanceCounter.tryLuck()
    }
}
